// prettier-plugin-java/dist/test/unit-test/snippets/arrays/array-initializer.spec.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var test_utils_1 = require("../../../test-utils");
describe("Arrays Initializer", function () {
    it("can format an empty arrayInitializer", function () {
        test_utils_1.expectSnippetToBeFormatted({
            snippet: "{ }",
            expectedOutput: "{}",
            entryPoint: "arrayInitializer"
        });
    });
    it("can format a arrayInitializer", function () {
        test_utils_1.expectSnippetToBeFormatted({
            snippet: "{alpha}",
            expectedOutput: "{ alpha }",
            entryPoint: "arrayInitializer"
        });
    });
    describe("Trailing Commas", function () {
        it("should remove extra comma in arrayInitializer by default", function () {
            test_utils_1.expectSnippetToBeFormatted({
                snippet: "{alpha,}",
                expectedOutput: "{ alpha }",
                entryPoint: "arrayInitializer"
            });
        });
        it("should remove extra comma in arrayInitializer by default", function () {
            var snippet = "{oneVeryLongArrayValue,}";
            var prettierOptions = {
                printWidth: 15
            };
            // prettier-ignore
            var expectedOutput = "{\n" +
                "  oneVeryLongArrayValue\n" +
                "}";
            test_utils_1.expectSnippetToBeFormatted({
                snippet: snippet,
                expectedOutput: expectedOutput,
                entryPoint: "arrayInitializer",
                prettierOptions: prettierOptions
            });
        });
        it("should remove extra comma in arrayInitializer if it fit in one line and --trailing-comma='all'", function () {
            test_utils_1.expectSnippetToBeFormatted({
                snippet: "{oneVeryLongArrayValue,}",
                expectedOutput: "{ oneVeryLongArrayValue }",
                entryPoint: "arrayInitializer",
                prettierOptions: {
                    trailingComma: "all"
                }
            });
        });
        it("should not add extra comma in arrayInitializer if it fit in one line and --trailing-comma='all'", function () {
            test_utils_1.expectSnippetToBeFormatted({
                snippet: "{oneVeryLongArrayValue}",
                expectedOutput: "{ oneVeryLongArrayValue }",
                entryPoint: "arrayInitializer",
                prettierOptions: {
                    trailingComma: "all"
                }
            });
        });
        it("should keep extra comma in arrayInitializer if it does not fit in one line and --trailing-comma='all'", function () {
            // prettier-ignore
            var expectedOutput = "{\n" +
                "  oneVeryLongArrayValue/* COMMA */,\n" +
                "}";
            test_utils_1.expectSnippetToBeFormatted({
                snippet: "{oneVeryLongArrayValue /* COMMA */,}",
                expectedOutput: expectedOutput,
                entryPoint: "arrayInitializer",
                prettierOptions: {
                    printWidth: 15,
                    trailingComma: "all"
                }
            });
        });
        it("should add extra comma in arrayInitializer if it does not fit in one line and --trailing-comma='all'", function () {
            // prettier-ignore
            var expectedOutput = "{\n" +
                "  oneVeryLongArrayValue,\n" +
                "}";
            test_utils_1.expectSnippetToBeFormatted({
                snippet: "{oneVeryLongArrayValue}",
                expectedOutput: expectedOutput,
                entryPoint: "arrayInitializer",
                prettierOptions: {
                    printWidth: 15,
                    trailingComma: "all"
                }
            });
        });
    });
});
