// prettier-plugin-java/dist/test/unit-test/snippets/classes/enum-body.spec.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var test_utils_1 = require("../../../test-utils");
describe("Enum Body", function () {
    it("Remove trailing comma by default", function () {
        var snippet = "{ONE,TWO,THREE,}";
        // prettier-ignore
        var expectedOutput = "{\n" +
            "  ONE,\n" +
            "  TWO,\n" +
            "  THREE\n" +
            "}";
        test_utils_1.expectSnippetToBeFormatted({
            snippet: snippet,
            expectedOutput: expectedOutput,
            entryPoint: "enumBody"
        });
    });
    it("Add trailing comma when specified", function () {
        var snippet = "{ONE,TWO,THREE}";
        // prettier-ignore
        var expectedOutput = "{\n" +
            "  ONE,\n" +
            "  TWO,\n" +
            "  THREE,\n" +
            "}";
        var prettierOptions = {
            trailingComma: "all"
        };
        test_utils_1.expectSnippetToBeFormatted({
            snippet: snippet,
            expectedOutput: expectedOutput,
            entryPoint: "enumBody",
            prettierOptions: prettierOptions
        });
    });
    it("Keep one trailing comma if there is already one", function () {
        var snippet = "{" + "\n  ONE," + "\n  TWO," + "\n  THREE," + "\n}";
        // prettier-ignore
        var expectedOutput = "{\n" +
            "  ONE,\n" +
            "  TWO,\n" +
            "  THREE,\n" +
            "}";
        var prettierOptions = {
            trailingComma: "all"
        };
        test_utils_1.expectSnippetToBeFormatted({
            snippet: snippet,
            expectedOutput: expectedOutput,
            entryPoint: "enumBody",
            prettierOptions: prettierOptions
        });
    });
    it("Remove trailing comma in enum constant list with semicolon by default", function () {
        var snippet = "{ONE,TWO,THREE,;}";
        // prettier-ignore
        var expectedOutput = "{\n" +
            "  ONE,\n" +
            "  TWO,\n" +
            "  THREE\n" +
            "}";
        test_utils_1.expectSnippetToBeFormatted({
            snippet: snippet,
            expectedOutput: expectedOutput,
            entryPoint: "enumBody"
        });
    });
    it("Add trailing comma in enum constant list with semicolon when specified", function () {
        var snippet = "{ONE,TWO,THREE;}";
        // prettier-ignore
        var expectedOutput = "{\n" +
            "  ONE,\n" +
            "  TWO,\n" +
            "  THREE,\n" +
            "}";
        var prettierOptions = {
            trailingComma: "all"
        };
        test_utils_1.expectSnippetToBeFormatted({
            snippet: snippet,
            expectedOutput: expectedOutput,
            entryPoint: "enumBody",
            prettierOptions: prettierOptions
        });
    });
});
